### DISCUSSION TITLE
Migrating to @vercel/otel for OpenTelemetry challenges

### DISCUSSION THREAD

**Discussion by chris-erickson** [vercel/next.js#65536] [Help]
### Summary

When running locally, everything is instrumented and the traces appear in our preferred observability service (Honeycomb). However, when I deploy the site to Vercel, no data arrives, and there's nothing in the runtime logs that really points to any particular issue.

We set our config via the environment like:

```
OTEL_EXPORTER_OTLP_ENDPOINT=https://api.honeycomb.io
OTEL_EXPORTER_OTLP_HEADERS=x-honeycomb-team=REDACTED,x-honeycomb-dataset=REDACTED
OTEL_DEFAULT_SAMPLE_PERCENTAGE=1
OTEL_LOG_LEVEL=debug
```

I've verified the same settings are on the preview environments.

We've used a completely custom initialization of the Otel stuff for quite a while now, but I want to upgrade to the `@vercel/otel` so that it also works in the app router (and is generally less to maintain). We're still using the older custom setup on production, so something between these two setups must be different (perhaps how `@vercel/otel` behaves on Vercel infra?).

What ideas are there for debugging this? Have people gotten `@vercel/otel` working well across pages and app router setups and shipping to a third-party store (as opposed to one of the integrated ones like Datadog).

### Additional information

```js
Current config:

import { registerOTel } from '@vercel/otel';
import type { Configuration } from '@vercel/otel';

const otelConfig: Configuration = {
  serviceName: 'Our Site - NEW',
};

export function register() {
  registerOTel(otelConfig);
}
```

**Comment by 404answernotfound** [comment]
Reading from here (and I'm assuming you're hosting your service on Vercel), maybe you need to add the same env variables to your Vercel configuration as well?

**Reply by chris-erickson** [reply]
Yep
> I've verified the same settings are on the preview environments.
